### PR TITLE
ChuckM stm32f429 examples

### PR DESCRIPTION
**Pull request by esden**
This pull requests contains Chucks stm32f429 discovery board examples.

I have made some minor improvements and reorganization to his additions.

This pull request is meant to discuss what needs to be done before we can merge this into our examples repository.

### CODE REVIEW

File: examples/stm32/f4/stm32f429i-discovery/sdram/console.c [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+ */
+void usart1_isr(void) {
+	uint32_t	reg;
+	int			i;
+
+	do {
+		reg = USART_SR(CONSOLE_UART);
+		if (reg & USART_SR_RXNE) {
+			recv_buf[recv_ndx_nxt] = USART_DR(CONSOLE_UART);
+#ifdef RESET_ON_CTRLC
```
Comment (ChuckM) [inline examples/stm32/f4/stm32f429i-discovery/sdram/console.c]: This should be updated to the "current" code which includes scb.h and calls scb_system_reset() which works correctly now.

**Comment by ChuckM** [comment]
So the general consensus is 
```
type func(param)
{
   ...
}
```

Rather than
```
type
func(params) {
   ...
}
```
Right? The old Sun kernel style put the function name in column 0 so that searching for ^name got you to the function rather than the first prototype if found but I can work with either style.

**Comment by ChuckM** [comment]
The console ^C handling code should be:
```
        /*
         * This bit of code will jump to the ResetHandler if you
         * hit ^C
         */
        if (recv_buf[nxt_recv_ndx] == '\03') {
            scb_reset_system();
            return; /* not actually reached */
        }
```

**Comment by esden** [comment]
@ChuckM Regarding code style, we use almost completely the linux kernel coding style. The make check target is using the kernel patch style check script: https://github.com/libopencm3/libopencm3/blob/master/scripts/checkpatch.pl https://github.com/torvalds/linux/blob/master/Documentation/CodingStyle

The check patch script should probably be updated to contain newer changes from the upstream kernel, and we should remove warnings about msleep < 20ms as that does not apply to us. But that is not part of this pull request. :)

**Comment by esden** [comment]
@ChuckM Corrected the CTRL-C handler as you described.

**Comment by esden** [comment]
Let me know guys, is this ok to be merged now or am I missing something?

**Comment by ChuckM** [comment]
:+1: from me.

**Pull request merged**